cms/search/signal_handlers.py

```diff
@@ -21,35 +19,50 @@ def get_publisher() -> KafkaPublisher | LogPublisher:
 
 
 @receiver(page_published)
-def on_page_published(sender: "Page", instance: "Page", **kwargs: dict) -> None:  # pylint: disable=unused-argument
+def on_page_published(sender: "type[Page]", instance: "Page", **kwargs: dict) -> None:  # pylint: disable=unused-argument
     """Called whenever a Wagtail Page is published (UI or code).
     instance is the published Page object.
     """
-    if instance.specific_class.__name__ not in SEARCH_INDEX_EXCLUDED_PAGE_TYPES:
+    if instance.specific_class.__name__ not in settings.SEARCH_INDEX_EXCLUDED_PAGE_TYPES:
         get_publisher().publish_created_or_updated(instance)
 
 
 @receiver(page_unpublished)
-def on_page_unpublished(sender: "Page", instance: "Page", **kwargs: dict) -> None:  # pylint: disable=unused-argument
+def on_page_unpublished(sender: "type[Page]", instance: "Page", **kwargs: dict) -> None:  # pylint: disable=unused-argument
     """Called whenever a Wagtail Page is unpublished (UI or code).
     instance is the unpublished Page object.
     """
     if (
         settings.CMS_SEARCH_NOTIFY_ON_DELETE_OR_UNPUBLISH
-        and instance.specific_class.__name__ not in SEARCH_INDEX_EXCLUDED_PAGE_TYPES
+        and instance.specific_class.__name__ not in settings.SEARCH_INDEX_EXCLUDED_PAGE_TYPES
     ):
         get_publisher().publish_deleted(instance)
 
 
 @receiver(post_delete, sender=Page)
-def on_page_deleted(sender: "Page", instance: "Page", **kwargs: dict) -> None:  # pylint: disable=unused-argument
+def on_page_deleted(sender: "type[Page]", instance: "Page", **kwargs: dict) -> None:  # pylint: disable=unused-argument
     """Catches all subclass deletions of Wagtail's Page model.
     Only fires if the page is published and not in SEARCH_INDEX_EXCLUDED_PAGE_TYPES.
     """
     # Only proceed if `sender` is a subclass of Wagtail Page and the page is published
     if (
         settings.CMS_SEARCH_NOTIFY_ON_DELETE_OR_UNPUBLISH
         and instance.live
-        and instance.specific_class.__name__ not in SEARCH_INDEX_EXCLUDED_PAGE_TYPES
+        and instance.specific_class.__name__ not in settings.SEARCH_INDEX_EXCLUDED_PAGE_TYPES
     ):
         get_publisher().publish_deleted(instance)
+
+
+@receiver(post_page_move)
+def on_page_moved(sender: "type[Page]", instance: "Page", **kwargs: dict) -> None:  # pylint: disable=unused-argument
+    """Called whenever a Wagtail Page is moved in the tree (UI or code).
+    instance is the moved Page object.
+    We use the publish_created_or_updated method to update search for the moved page and any of it's non-excluded
+    descendants, which will also be affected by the move.
+    """
```

Comment:
Probably best as a follow-up, it just occurred to me that we also need to clear the cache. Will raise a ticket.